### PR TITLE
test: Add more unit tests for `add_to_list`.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2531)

### CODE REVIEW

File: toxcore/DHT_test.cc
Comment (Green-Sky) [inline toxcore/DHT_test.cc:213]: make deterministic before merge

**Comment by nurupo** [comment]
*[`toxcore/crypto_core_test_util.cc` line 51 at r2](https://reviewable.io/reviews/TokTok/c-toxcore/2531#-Nnkqpzf9ny6WhbYmBjv:-Nnkqpzf9ny6WhbYmBjw:b2jgqkv) ([raw file](https://github.com/TokTok/c-toxcore/blob/9f57460e30375cc192793c9863cd1f5f94f36613/toxcore/crypto_core_test_util.cc#L51)):*
> ```cpp
> {
>     Test_Random *self = static_cast<Test_Random *>(obj);
>     return static_cast<uint32_t>(simple_rng(self->seed)) % upper_bound;
> ```

Nit: modulo doesn't result in a uniform distribution, it gets skewed. Might not matter for testing though.
C++' random library has random number generators and`std::uniform_int_distribution()` too, wonder why they weren't used here.

**Review by iphydf — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2531)* status: 0 of 1 approvals obtained (waiting on @Green-Sky and @nurupo)

___
*[`toxcore/crypto_core_test_util.cc` line 51 at r2](https://reviewable.io/reviews/TokTok/c-toxcore/2531#-Nnkqpzf9ny6WhbYmBjv:-NnkzB6G3Nz2GViCGveh:b8vt3s) ([raw file](https://github.com/TokTok/c-toxcore/blob/9f57460e30375cc192793c9863cd1f5f94f36613/toxcore/crypto_core_test_util.cc#L51)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

Nit: modulo doesn't result in a uniform distribution, it gets skewed. Might not matter for testing though.
C++' random library has random number generators and`std::uniform_int_distribution()` too, wonder why they weren't used here.
</blockquote></details>

Added a comment. I'll test/benchmark the C++ stdlib stuff later. For now, this is a very simple and fast implementation intended to be exactly that.

**Review by iphydf — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2531)* status: 0 of 1 approvals obtained (waiting on @Green-Sky and @nurupo)

___
*[`toxcore/crypto_core_test_util.cc` line 51 at r2](https://reviewable.io/reviews/TokTok/c-toxcore/2531#-Nnkqpzf9ny6WhbYmBjv:-Nnkzg_F8eU0kCI1ju5m:bfg5j4u) ([raw file](https://github.com/TokTok/c-toxcore/blob/9f57460e30375cc192793c9863cd1f5f94f36613/toxcore/crypto_core_test_util.cc#L51)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

Added a comment. I'll test/benchmark the C++ stdlib stuff later. For now, this is a very simple and fast implementation intended to be exactly that.
</blockquote></details>

I guess LCG is the same as I put above. Maybe I'll refactor it tomorrow, or otherwise it'll be done in the next round of refactorings.

**Review by iphydf — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2531)* status: 0 of 1 approvals obtained (waiting on @Green-Sky and @nurupo)

___
*[`toxcore/crypto_core_test_util.cc` line 51 at r2](https://reviewable.io/reviews/TokTok/c-toxcore/2531#-Nnkqpzf9ny6WhbYmBjv:-Nnl1higEm7UxCMHrftt:b-tkkeqt) ([raw file](https://github.com/TokTok/c-toxcore/blob/9f57460e30375cc192793c9863cd1f5f94f36613/toxcore/crypto_core_test_util.cc#L51)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

I guess LCG is the same as I put above. Maybe I'll refactor it tomorrow, or otherwise it'll be done in the next round of refactorings.
</blockquote></details>

I've decided to use C++ :P. Done.

**Review by nurupo — COMMENTED** [review]
> Reviewed 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2531)* status: 0 of 1 approvals obtained (waiting on @Green-Sky)

**Comment by nurupo** [comment]
FYI was just pointing out something I noticed while skimming through the PR, don't plan on seriously reviewing it so don't hold the PR waiting on my review approval.

**Review by Green-Sky — APPROVED** [review]

**Pull request merged**
